plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/ExMemoryUsage.hs

```diff
@@ -258,14 +258,35 @@ instance ExMemoryUsage Data where
             I n        -> memoryUsage n
             B b        -> memoryUsage b
 
+
+{- Note [Costing constant-size types]
+The memory usage of each of the BLS12-381 types is constant, so we may be able
+to optimise things a little by ensuring that we don't re-compute the size of
+(say) a G1 element every time one is used. GHC will probably do this anyway, but
+we make sure by defining a top level function for each of the size measures and
+getting the memoryUsage instances to call those.
+-}
+
```

Comment:
The stuff above and below is in response to https://github.com/input-output-hk/plutus/pull/5231#discussion_r1192712631 from @effectfully .